group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3852,6 +3844,27 @@ public static String consumerGroupRebalanceTimeoutKey(String groupId, String mem
         return "rebalance-timeout-" + groupId + "-" + memberId;
     }
 
+    public static String classicGroupSizeCounterKey() {
+        return "classic-group-size-counter";
+    }
+
+    /**
+     * Schedules (or reschedules) the group size counter for the classic group.
+     */
+    private void scheduleClassicGroupSizeCounter() {
```

Comment:
At the moment, I think that all the scheduled tasks are defined in the shard. I wonder if we could do the same for this one?

Reply:
I'm not sure how to move it to the shard. Since we don't want to pass `groups` around and we directly pass `Map<ClassicGroupState, Long>`, I feel that the process of counting groups has to be in `GroupMetadataManager`.

Reply:
I suppose that both are fines. Take a look at `scheduleGroupMetadataExpiration` in the shard. It would work the same.